### PR TITLE
pipeline api: add volumClaimTemplate

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind api-change

**What this PR does**:

Add complete pvc config about `SharedWorkspace`，rather than a string.

**Why we need it**:

Using a pre-created PVC (Persistent Volume Claim) in a pipeline then references it name is not recommended, as PVCs are meant to be linked with **each specific pipeline execution instance**, not the pipeline itself.

To simplify the user experience, only fields that have appeared in the Tekton Hub are included.

In total, there are only four configurations, among which **almost only the `RequestsStorage` is a meaningful setting**. The remaining three, even if configured, are almost always default settings.

### CODE REVIEW

File: pkg/apis/pipeline/v1alpha1/type.go [hunk 1]
```diff
@@ -46,10 +46,32 @@ type PipelineSpec struct {
...
-	SharedWorkspace *string `json:"sharedWorkspace,omitempty"`
+	SharedWorkspace *VolumeClaimTemplate `json:"sharedWorkspace,omitempty"`
+}
+
+// VolumeClaimTemplate is the configuration for the volume claim template in pipeline execution.
+// For more details, see https://github.com/kubernetes/api/blob/master/core/v1/types.go
+type VolumeClaimTemplate struct {
+	// AccessMode determines the access modes for the volume, e.g., ReadWriteOnce.
+	// This affects how the volume can be mounted.
+	AccessMode string `json:"accessMode,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/pipeline/v1alpha1/type.go:62]: Donot use string, please refer to k8s https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

And its API: `AccessModes []PersistentVolumeAccessMode` , PersistentVolumeAccessMode is a type aliase

File: pkg/apis/pipeline/v1alpha1/type.go [hunk 2]
```diff
@@ -46,10 +46,32 @@ type PipelineSpec struct {
...
+
+	// RequestsStorage defines the storage size required for this PVC, e.g., 1Gi, 100Mi.
+	// It specifies the storage capacity needed as part of ResourceRequirements.
+	RequestsStorage string `json:"requestsStorage,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/pipeline/v1alpha1/type.go:66]: How do you validate `string` with `Gi`, `Mi`

Comment (hzxuzhonghu) [inline pkg/apis/pipeline/v1alpha1/type.go:66]: s/RequestsStorage/StorageRequest, it is singular

File: pkg/apis/pipeline/v1alpha1/type.go [hunk 3]
```diff
@@ -46,10 +46,32 @@ type PipelineSpec struct {
...
+
+	// StorageClassName specifies the StorageClass name to which this persistent volume belongs, e.g., manual.
+	// It allows the PVC to use the characteristics defined by the StorageClass.
+	StorageClassName string `json:"storageClassName,omitempty"`
+
+	// VolumeMode specifies whether the volume should be used with a formatted filesystem (Filesystem)
+	// or remain in raw block state (Block). The Filesystem value is implied when not included.
+	VolumeMode string `json:"volumeMode,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/pipeline/v1alpha1/type.go:74]: Same as above donot use a generic string type

File: pkg/apis/pipeline/v1alpha1/type.go (hunk 1 above)
Comment (Xieql) [reply pkg/apis/pipeline/v1alpha1/type.go:62]: ok

File: pkg/apis/pipeline/v1alpha1/type.go (hunk 2 above)
Comment (Xieql) [reply pkg/apis/pipeline/v1alpha1/type.go:66]: we can use Pattern like // +kubebuilder:validation:Pattern="^[0-9]+(\\.[0-9]+)?(Gi|Mi)$"

Comment (Xieql) [reply pkg/apis/pipeline/v1alpha1/type.go:66]: ok

**Comment by Xieql** [comment]
@hzxuzhonghu PTAL

File: pkg/apis/pipeline/v1alpha1/type.go [hunk 4]
```diff
@@ -46,10 +46,39 @@ type PipelineSpec struct {
...
 	Tasks []PipelineTask `json:"tasks"`
 
-	// SharedWorkspace is the name of the PVC. If not specified, a PVC with the Pipeline's name as prefix will be created by default.
+	// SharedWorkspace is the config of the PVC where task using
+	// The PersistentVolumeClaim with this config will be created for each pipeline execution
+	// it allows the user to specify e.g. size and StorageClass for the volume.
 	// If not set, Kurator will create a PVC named Pipeline.name using default config
 	// +optional
-	SharedWorkspace *string `json:"sharedWorkspace,omitempty"`
+	SharedWorkspace *VolumeClaimTemplate `json:"sharedWorkspace,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/pipeline/v1alpha1/type.go:54]: I did not realize: isn't this a PVC spec. why do we redefine a new one?

Comment (Xieql) [reply pkg/apis/pipeline/v1alpha1/type.go:54]: Using a pre-created PVC (Persistent Volume Claim) in a pipeline then references it name is not recommended, as PVCs are meant to be linked with each specific pipeline execution instance, not the pipeline itself.

**Pull request merged**